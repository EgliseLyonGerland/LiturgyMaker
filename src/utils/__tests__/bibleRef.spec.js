const superagent = require('superagent');
const { getPassage } = require('../bibleRef');
const getbibleContent = require('./getbible.json');

jest.mock('superagent');

superagent.end = jest.fn().mockImplementation(() => ({
  body: { data: { content: '' } },
}));

beforeEach(() => {
  jest.clearAllMocks();
});

test('getPassage() with "1 Pierre 1"', async () => {
  await getPassage('1 Pierre 1');

  expect(superagent.query).toHaveBeenCalledWith({
    passage: '1 Pe 1',
    version: 'darby',
  });
});

test('getPassage() with "1 Pierre 1-2"', async () => {
  await getPassage('1 Pierre 1-2');

  expect(superagent.query).toHaveBeenCalledWith({
    passage: '1 Pe 1:1-999;2:1-999',
    version: 'darby',
  });
});

test('getPassage() with "1 Pierre 1-4"', async () => {
  await getPassage('1 Pierre 1-4');

  expect(superagent.query).toHaveBeenCalledWith({
    passage: '1 Pe 1:1-999;2:1-999;3:1-999;4:1-999',
    version: 'darby',
  });
});

test('getPassage() with "1 Pierre 1.1"', async () => {
  await getPassage('1 Pierre 1.1');

  expect(superagent.query).toHaveBeenCalledWith({
    passage: '1 Pe 1:1',
    version: 'darby',
  });
});

test('getPassage() with "1 Pierre 1.1-4"', async () => {
  await getPassage('1 Pierre 1.1-4');

  expect(superagent.query).toHaveBeenCalledWith({
    passage: '1 Pe 1:1-4',
    version: 'darby',
  });
});

test('getPassage() with "1 Pierre 1.1-2.2"', async () => {
  await getPassage('1 Pierre 1.1-2.2');

  expect(superagent.query).toHaveBeenCalledWith({
    passage: '1 Pe 1:1-999;2:1-2',
    version: 'darby',
  });
});

test('getPassage() with "1 Pierre 1.1-4.4"', async () => {
  await getPassage('1 Pierre 1.1-4.4');

  expect(superagent.query).toHaveBeenCalledWith({
    passage: '1 Pe 1:1-999;2:1-999;3:1-999;4:1-4',
    version: 'darby',
  });
});

test('stringifyContent()', async () => {
  superagent.end = jest.fn().mockImplementation(() => ({
    body: { data: { content: getbibleContent } },
  }));

  const result = await getPassage('1 Pierre 1.1-4.4');

  expect(result).toMatchSnapshot();
});
