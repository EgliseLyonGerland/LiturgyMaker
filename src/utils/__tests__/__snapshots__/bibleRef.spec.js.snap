// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringifyContent() 1`] = `""`;
